### PR TITLE
Add memory metering for `Int8`, `Int16`, `Int32` and `Int64` Values

### PR DESCRIPTION
**Pull request by SupunS**
Work towards https://github.com/dapperlabs/cadence-private-issues/issues/2

## Description

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/decode.go [hunk 1]
```diff
@@ -350,6 +350,8 @@ func (d StorableDecoder) decodeInt() (IntValue, error) {
 }
 
 func (d StorableDecoder) decodeInt8() (Int8Value, error) {
+	common.UseMemory(d.memoryGauge, int8MemoryUsage)
+
 	v, err := d.decoder.DecodeInt64()
```
Comment (turbolent) [inline runtime/interpreter/decode.go:355]: As a `int64` is decoded, it should be metered as such. Maybe define a function `decodeInt64` which meters 8 bytes and calls the `StreamDecoder.DecodeInt64`, e.g. like the `decodeString` helper function

**Review by turbolent — COMMENTED** [review]
Nice, thank you!

**Review by robert-e-davidson3 — APPROVED** [review]

File: runtime/interpreter/decode.go (hunk 1 above)
Comment (SupunS) [reply runtime/interpreter/decode.go]: added with e76c1f6

**Review by turbolent — APPROVED** [review]
Great work! 👍

**Pull request merged**